modules/shortcodes/wordads.php

```diff
@@ -53,14 +57,41 @@ static function wordads_shortcode_html( $atts, $content = '' ) {
 			return '<div>' . __( 'The WordAds module is not active', 'jetpack' ) . '</div>';
 		}
 
-		$html = '<div class="jetpack-wordad" itemscope itemtype="https://schema.org/WPAdBlock">';
-
-		$html .= '</div>';
-
+		$html = '<div class="jetpack-wordad" itemscope itemtype="https://schema.org/WPAdBlock"></div>';
 		$html = $wordads->insert_inline_ad( $html );
 
 		return $html;
 	}
+
+	public static function gutenblock_render( $attr ) {
+		global $wordads;
+
+		/** This filter is already documented in modules/wordads/wordads.php `insert_ad()` */
+		if ( is_feed() || apply_filters( 'wordads_inpost_disable', false ) ) {
+			return '';
+		}
+
+		if ( $wordads->option( 'wordads_house' ) ) {
+			return $wordads->get_ad( 'inline', 'house' );
+		}
+
+		$section_id = 0 === $wordads->params->blog_id ? WORDADS_API_TEST_ID : $wordads->params->blog_id . '6'; // 6 is to keep track of gutenblock ads
+		$align = 'center';
+		if ( isset( $attr['align'] ) && in_array( $attr['align'], array( 'left', 'center', 'right' ) ) ) {
+			$align = $attr['align'];
+		}
+		$align = 'align' . $align;
+
+		$format = 'mrec';
+		if ( isset( $attr['format'] ) && in_array( $attr['format'], array_keys( WordAds::$ad_tag_ids ) ) ) {
+			$format = $attr['format'];
+		}
+
+		$height = WordAds::$ad_tag_ids[$format]['height'];
+		$width = WordAds::$ad_tag_ids[$format]['width'];
+		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', 'float:left;margin-right:5px;margin-top:0px;' );
```

Comment:
😮 This method call makes me a bit uncomfortable 😬 

We're reproducing this?

https://github.com/Automattic/jetpack/blob/bdeb56a72746d4d00910259b1a87011e89bc7786/modules/wordads/wordads.php#L425

The reason it's duplicated is to be able to assign a different suffix? (6 not 5)?

https://github.com/Automattic/jetpack/blob/bdeb56a72746d4d00910259b1a87011e89bc7786/modules/wordads/wordads.php#L424

The partial style attribute string, especially when duplicated in two places, seems like an easy thing to screw up and get out of sync in later work. All of the other use cases for the final param are at least side by side:

https://github.com/Automattic/jetpack/blob/4f6c69e8c8d7406a68dcbb4d16f6bc17e18fc0d7/modules/wordads/wordads.php#L418-L425

Do you think we could change this so that at least we're not leaking that into this file as well?

Reply:
This extra bit of css has to do with the special nature of the fixed locations the other ad units get inserted into (header/belowpost/sidebar/etc). Sometimes they require a little bit of extra css to ensure they slip in nicely above/below/beside other ad units and whatnot as well as have enough space for the "Report This Ad" link. Their locations/format are fixed and unlikely to change, but I still wanted a function that was generic enough to get reused in places.

Basically there's a lot of extra "stuff" going on in the original function that's not really needed for what we're doing in the block, so I wanted to streamline/simplify it. It's probable in the future that we'll remove the extra CSS bit entirely as we update our external system providing the ads to be smarter about inserting themselves.

Reply:
What about some middle ground like moving the shared attribute style string into an appropriately named class constant and actually sharing it?

```suggestion
		$snippet = $wordads->get_ad_snippet( $section_id, $height, $width, 'inline', WordAds::DAT_UGLY_CSS_WE_GOTTA_SHARE );
```

Reply:
Sure thing, added `public static $SOLO_UNIT_CSS = 'float:left;margin-right:5px;margin-top:0px;';`